internal/providers/azure.go

```diff
@@ -0,0 +1,59 @@
+package providers
+
+import (
+	"log"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/google/uuid"
+)
+
+type AzureProvider struct {
+	Credentials         *azidentity.DefaultAzureCredential
+	ClientOptions       *arm.ClientOptions
+	SubscriptionID      string
+	UniqueCorrelationId uuid.UUID // unique value injected in the user-agent, to differentiate Stratus Red Team executions
+}
+
+var DefaultClientOptions = arm.ClientOptions{
+	ClientOptions: azcore.ClientOptions{
+		Telemetry: policy.TelemetryOptions{ApplicationID: UniqueExecutionId.String(), Disabled: false},
+	},
+}
+
+var azureProvider = AzureProvider{
+	UniqueCorrelationId: UniqueExecutionId,
+	SubscriptionID:      os.Getenv("AZURE_SUBSCRIPTION_ID"),
```

Comment:
It would be better if this could be inferred based on the context (i.e. based on your Azure CLI context set through `az account set --subscription xyz`).

Since this subscription ID seems to only be used in `IsAuthenticatedAgainstAzure`, it looks like we could remove it and use `azidentity.NewDefaultAzureCredential(nil)` everywhere? (which authenticates based on context)

Reply:
Discussed over DM and in other threads. There is no convenient way to do this through the AzureCLI's profile.

We also want the subscriptionID to remain in the provider for easy access during detonation. Otherwise we would need to obtain it in each `detonate` function.